actions/registry

```diff
@@ -1,152 +0,0 @@
-# run will create/delete the registry, and the reset will be a no-op and won't
-# error out)
-#
-# - take only a plain authentication file, and create the encrypted version in
-# the action
-#
-# - validate the parameters (make sure tlscert is a certificate, that tlskey is a
-# proper key, etc)
-#
-# - when https://bugs.launchpad.net/juju/+bug/1661015 is fixed, handle the
-# base64 encoding the parameters in the action itself
-
-import os
-import sys
-
-from base64 import b64encode
-
-from charmhelpers.core.hookenv import action_get
-from charmhelpers.core.hookenv import action_set
-from charms.templating.jinja2 import render
-from charms.reactive import endpoint_from_flag
-from subprocess import call, check_output
-
-os.environ['PATH'] += os.pathsep + os.path.join(os.sep, 'snap', 'bin')
-
-deletion = action_get('delete')
-
-context = {}
-
-arch = check_output(['dpkg', '--print-architecture']).rstrip()
-context['arch'] = arch.decode('utf-8')
-
-# This action was deprecated in 1.17.
-action_set({
-    'notice':
-    ('DEPRECATED: See https://ubuntu.com/kubernetes/docs/docker-registry '
-     'for supported container registry options.')
-})
-
-# These config options must be defined in the case of a creation
-param_error = False
-for param in ('tlscert', 'tlskey', 'domain', 'htpasswd', 'htpasswd-plain'):
-    value = action_get(param)
-    if not value and not deletion:
-        key = "registry-create-parameter-{}".format(param)
-        error = "failure, parameter {} is required".format(param)
-        action_set({key: error})
-        param_error = True
-
-    context[param] = value
-
-# Create the dockercfg template variable
-dockercfg = '{"%s": {"auth": "%s", "email": "root@localhost"}}' % \
-            (context['domain'], context['htpasswd-plain'])
-context['dockercfg'] = b64encode(dockercfg.encode()).decode('ASCII')
-
-if param_error:
-    sys.exit(0)
-
-# This one is either true or false, no need to check if it has a "good" value.
-context['ingress'] = action_get('ingress')
-
-# Declare a kubectl template when invoking kubectl
-kubectl = ['kubectl', '--kubeconfig=/root/.kube/config']
-
-# Remove deployment if requested
-if deletion:
-    resources = ['svc/kube-registry', 'rc/kube-registry-v0', 'secrets/registry-tls-data',
-                 'secrets/registry-auth-data', 'secrets/registry-access']
-
-    if action_get('ingress'):
-        resources.append('ing/registry-ing')
-
-    delete_command = kubectl + ['delete', '--ignore-not-found=true'] + resources
-    delete_response = call(delete_command)
-    if delete_response == 0:
-        action_set({'registry-delete': 'success'})
-    else:
-        action_set({'registry-delete': 'failure'})
-
-    sys.exit(0)
-
-kube_control = endpoint_from_flag('kube-control.registry_location.available')
-if kube_control:
-    registry_location = kube_control.get_registry_location()
-    context['registry'] = registry_location
-
-# Creation request
-render('registry.yaml', '/root/cdk/addons/registry.yaml',
```

Comment:
I see no other references to registry.yaml, so I think you can remove `templates/registry.yaml` too!